Sources/ReadingListKit/PocketAuthenticationService.swift

```diff
@@ -0,0 +1,106 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import AuthenticationServices
+import Combine
+
+struct Response {
+    let guid: String
+    let accessToken: String
+    let userIdentifier: String
+}
+
+enum PocketAuthenticationError: Error {
+    case invalidRedirect
+    case invalidComponents
+    case alreadyAuthenticating
+    case noConsumerKey
+    case other(Swift.Error)
+}
+
+class PocketAuthenticationService: NSObject, ASWebAuthenticationPresentationContextProviding {
+    private var isAuthenticating = false
+
+    @MainActor
+    func logIn() async throws -> Response {
+        defer { isAuthenticating = false }
+
+        guard isAuthenticating == false else {
+            throw PocketAuthenticationError.alreadyAuthenticating
+        }
+
+        isAuthenticating = true
+        return try await authenticate(with: "/login", contextProvider: self)
+    }
+
+    @MainActor
+    func signUp() async throws -> Response {
+        defer { isAuthenticating = false }
+
+        guard isAuthenticating == false else {
+            throw PocketAuthenticationError.alreadyAuthenticating
+        }
+
+        isAuthenticating = true
+        return try await authenticate(with: "/signup", contextProvider: self)
+    }
+
+    private func authenticate(with path: String, contextProvider: ASWebAuthenticationPresentationContextProviding?) async throws -> Response {
+        guard var components = URLComponents(string: "https://getpocket.com") else {
+            throw PocketAuthenticationError.invalidComponents
+        }
+
+        let requestRedirect = "pocket"
+        let consumerKey: String? = nil // TODO: Store consumerKey somewhere off Github. Fetch when needed.
```

Comment:
Hardcoding the consumer key in the code might not be desirable, since this file is committed, and for code cleanliness. A better strategy seems to be what we do in Pocket, what do you think?

Reply:
Those are exactly my thoughts also!
We don't currently have a way to handle secrets in the app (That I can see), and it seems out of scope for this pr, so I just cheated. Basically.

Reply:
I will have a PR to add secrets today. Previously, we were using both iPad and iPhone consumer keys, but will only add one consumer key for simplicity sake on this codebase.